plugin/storage/es/factory.go

```diff
@@ -403,6 +405,18 @@ func (f *Factory) onClientPasswordChange(cfg *config.Configuration, client *atom
 	}
 }
 
+func (f *Factory) Purge() error {
+	ctx := context.Background()
```

Comment:
this is an anti-pattern - I think we should change Purge to accept ctx as argument.